src/super_gradients/training/models/segmentation_models/stdc.py

```diff
@@ -35,13 +35,17 @@ def __init__(self, in_channels: int, out_channels: int, steps: int, stdc_downsam
         """
         super().__init__()
         assert steps in [2, 3, 4], f"only 2, 3, 4 steps number are supported, found: {steps}"
```

Comment:
This is not directly related to changes of this PR, but since you're already touching the c'tor - please replace assert statement with if / raise statement.

**Why: Assertions can be toggled on or off at will.**

https://stefan.sofa-rockers.org/2018/04/16/assertions-and-exceptions/

> So never use an assert when another method of capturing and raising an error is possible. Always write your code with the assumption that the assertions could disappear at any minute. And if you’re unsure, you can run your program with python -O or by setting PYTHONOPTIMIZE=1 to test if it still works.

TLDR: assert in unit tests, if/raise in library code